core-processor/src/processor.rs

```diff
@@ -37,12 +40,19 @@ pub fn process<A: ProcessorExt + EnvExt + Into<ExtInfo> + 'static, E: Environmen
     dispatch: Dispatch,
     block_info: BlockInfo,
     existential_deposit: u128,
+    initiator: ProgramId,
 ) -> Vec<JournalNote> {
     if let Some(actor) = actor {
         let execution_settings = ExecutionSettings::new(block_info, existential_deposit);
+        let execution_context = ExecutionContext { initiator };
```

Comment:
Can we combine settings and context into one entity?

Reply:
I don't think so.
Setting generate at the start of the queue processing and don't change from message to message inside the same block, while context relates to each message separately and will contain more data in future, e.g. available gas

Reply:
Got it, ok.